server/routerlicious/packages/lambdas/src/deli/lambda.ts

```diff
@@ -827,9 +856,52 @@ export class DeliLambda implements IPartitionLambda {
     }
 
     private clearNoopConsolidationTimer() {
-        if (this.noopTimer !== undefined) {
-            clearTimeout(this.noopTimer);
-            this.noopTimer = undefined;
+        if (this.noopEvent !== undefined) {
+            clearTimeout(this.noopEvent);
+            this.noopEvent = undefined;
+        }
+    }
+
+    private updateOpIdleTimer() {
+        this.clearOpIdleTimer();
+
+        this.opIdleTimer = setTimeout(() => {
+            this.emitOpEvent("idle");
+        }, this.serviceConfiguration.deli.opEvent.idleTime);
+    }
+
+    private clearOpIdleTimer() {
+        if (this.opIdleTimer !== undefined) {
+            clearTimeout(this.opIdleTimer);
+            this.opIdleTimer = undefined;
+        }
+    }
+
+    private updateOpMaxTimeTimer() {
+        this.clearOpMaxTimeTimer();
+
+        this.opMaxTimeTimer = setTimeout(() => {
+            this.emitOpEvent("maxTime");
+        }, this.serviceConfiguration.deli.opEvent.maxTime);
+    }
+
+    private clearOpMaxTimeTimer() {
+        if (this.opMaxTimeTimer !== undefined) {
+            clearTimeout(this.opMaxTimeTimer);
+            this.opMaxTimeTimer = undefined;
+        }
+    }
+
+    private emitOpEvent(type: string) {
```

Comment:
Should we restrict the type to these three strings?

Reply:
Yea. I changed it to an enum so it's strongly typed